source/gateway.go

```diff
@@ -224,21 +223,29 @@ func (sc *gatewaySource) setResourceLabel(config istiomodel.Config, endpoints []
 	}
 }
 
-func (sc *gatewaySource) targetsFromIstioIngressStatus() (targets endpoint.Targets, err error) {
-	if svc, e := sc.kubeClient.CoreV1().Services(sc.istioNamespace).Get(sc.istioIngressGatewayName, metav1.GetOptions{}); e != nil {
-		err = e
-	} else {
-		for _, lb := range svc.Status.LoadBalancer.Ingress {
-			if lb.IP != "" {
-				targets = append(targets, lb.IP)
-			}
-			if lb.Hostname != "" {
-				targets = append(targets, lb.Hostname)
+func (sc *gatewaySource) targetsFromIstioIngressGatewayServices() (targets endpoint.Targets, err error) {
+	for _, lbService := range sc.istioIngressGatewayServices {
+		lbNamespace, lbName, e := parseIngressGateway(lbService)
+		if e != nil {
+			err = e
+			return nil, err
+		}
+		if svc, e := sc.kubeClient.CoreV1().Services(lbNamespace).Get(lbName, metav1.GetOptions{}); e != nil {
+			err = e
+			log.Warn(err)
+		} else {
```

Comment:
I think we can omit the else clause.

Reply:
We could omit it if we returned on error before, but I'm just logging it, so it wouldn't be safe to still run the for loop in that case. Should I change to return on error? Otherwise I don't think it can be omitted here